ui/src/client/UpdateClientDialog.tsx

```diff
@@ -0,0 +1,83 @@
+import Button from '@material-ui/core/Button';
+import Dialog from '@material-ui/core/Dialog';
+import DialogActions from '@material-ui/core/DialogActions';
+import DialogContent from '@material-ui/core/DialogContent';
+import DialogContentText from '@material-ui/core/DialogContentText';
+import DialogTitle from '@material-ui/core/DialogTitle';
+import TextField from '@material-ui/core/TextField';
+import Tooltip from '@material-ui/core/Tooltip';
+import React, {Component} from 'react';
+
+interface IProps {
+    fClose: VoidFunction;
+    fOnSubmit: (name: string) => void;
+    initialName: string;
+}
+
+interface IState {
+    name: string;
+}
+
+export default class UpdateDialog extends Component<IProps, IState> {
+    public state = {name: ''};
+
+    public componentWillMount() {
+        this.setState({name: this.props.initialName});
+    }
+
+    public render() {
+        const {fClose, fOnSubmit} = this.props;
+        const {name} = this.state;
+        const submitEnabled = this.state.name.length !== 0;
+        const submitAndClose = () => {
+            fOnSubmit(name);
+            fClose();
+        };
+        return (
+            <Dialog
+                open={true}
+                onClose={fClose}
+                aria-labelledby="form-dialog-title"
+                id="client-dialog">
+                <DialogTitle id="form-dialog-title">Update a Client</DialogTitle>
+                <DialogContent>
+                    <DialogContentText>
+                        A client receives message and manages creation of a new tokens and delete
+                        messages.
+                    </DialogContentText>
+                    <TextField
+                        autoFocus
+                        margin="dense"
+                        className="name"
+                        label="Name *"
+                        type="text"
+                        value={name}
+                        onChange={this.handleChange.bind(this, 'name')}
+                        fullWidth
+                    />
+                </DialogContent>
+                <DialogActions>
+                    <Button onClick={fClose}>Cancel</Button>
+                    <Tooltip title={submitEnabled ? '' : 'name is required'}>
+                        <div>
+                            <Button
+                                className="update"
+                                disabled={!submitEnabled}
+                                onClick={submitAndClose}
+                                color="primary"
+                                variant="raised">
+                                Update
+                            </Button>
+                        </div>
+                    </Tooltip>
+                </DialogActions>
+            </Dialog>
+        );
+    }
+
+    private handleChange(propertyName: string, event: React.ChangeEvent<HTMLInputElement>) {
+        const state = this.state;
```

Comment:
```suggestion
        const state = {};
```
State is merged to the previous state automatically, so I think an empty object is enough. I don't have much react experience though, so feel free to appeal this if you think I am missing something:)